src/uct/ib/ud/base/ud_ep.c

```diff
@@ -781,15 +891,31 @@ static void uct_ud_ep_rx_creq(uct_ud_iface_t *iface, uct_ud_neth_t *neth)
                                  ctl->conn_req.path_index,
                                  ctl->conn_req.conn_sn, 0);
     if (ep == NULL) {
-        ep = uct_ud_ep_create_passive(iface, ctl);
-        if (ep == NULL) {
-            ucs_diag("iface %p: failed to create the passive endpoint", iface);
-            return;
-        }
+        if ((ctl->conn_req.dest_ep_id != UCT_UD_EP_NULL_ID) &&
+            ucs_ptr_array_lookup(&iface->eps, ctl->conn_req.dest_ep_id, ep)) {
```

Comment:
to check whether we have corresponding endpoint in PTR array or not.
it may happen that endpoint was destroy by use prior receiving CREQ from the peer - so checking this endpoint in PTR array helps to find such endpoits if they are not in CEP anymore.